xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-api/src/main/java/org/xwiki/notifications/script/NotificationScriptService.java

```diff
@@ -296,4 +302,66 @@ public String getFeed(String userId, int entryNumber, boolean onlyUnread) throws
             throw new NotificationException("Unable to render RSS feed", e);
         }
     }
+
+    /**
+     * Update a notification preference of the given user that matches the given eventType and notificationType to
+     * the given notificationStatus and the given startDate.
+     *
+     * @param eventType the type of the event
+     * @param applicationId the ID of the application concerned
+     * @param notificationFormat the format of the notification (see {@link NotificationFormat}
+     * @param notificationStatus the status (enabled / disabled) of the notification
+     * @param startDate the date from which the user should receive notifications of this type
+     * @throws NotificationException if an error occurs
+     *
+     * @since 9.7RC1
+     */
+    public void setNotificationPreferenceStatus(String eventType, String applicationId,
+            String notificationFormat, boolean notificationStatus,
+            Date startDate) throws NotificationException
+    {
+        NotificationFormat format = NotificationFormat.valueOf(notificationFormat.toUpperCase());
+
+        // Build the corresponding NotificationPreference object
+        NotificationPreference preference = new NotificationPreference(eventType, applicationId, notificationStatus,
+                format, startDate);
+
+        try {
+            this.modelBridge.saveNotificationPreference(documentAccessBridge.getCurrentUserReference(), preference);
+        } catch (NotificationException e) {
+            throw new NotificationException(String.format(
+                    "Unable to save the notification preferences for [%s] in [%s]", eventType, applicationId));
+        }
+    }
+
+    /**
+     * As {@link #setNotificationPreferenceStatus(String, String, String, boolean, Date)}, update the notification
+     * preference of a given user.
+     *
+     * @param userId the id of the user to edit
+     * @param eventType the type of the event
+     * @param applicationId the ID of the application concerned
+     * @param notificationFormat the format of the notification (see {@link NotificationFormat}
+     * @param notificationStatus the status (enabled / disabled) of the notification
+     * @param startDate the date from which the user should receive notifications of this type
+     * @throws NotificationException if an error occurs
+     *
+     * @since 9.7RC1
+     */
+    public void setNotificationPreferenceStatus(String userId, String eventType, String applicationId,
+            String notificationFormat, boolean notificationStatus, Date startDate) throws NotificationException
+    {
+        NotificationFormat format = NotificationFormat.valueOf(notificationFormat.toUpperCase());
```

Comment:
If you don't check the rights, everybody is able to change the preferences of everybody.